source/image-handler/index.ts

```diff
@@ -138,3 +139,23 @@ function getResponseHeaders(isError: boolean = false, isAlb: boolean = false): H
 
   return headers;
 }
+
+/**
+ * Transorm CDN URLs to the format expected by the image handler.
+ * @param url
+ * @returns Transformed URL.
+ */
+function transformCdnUrls(url: string): string {
+  // Regular expression to detect the second type of URL and capture the dynamic parts
+  const regex = /\/fit-in\/(\d+x\d+)\/filters:quality\((\d+)\)\/cdn-cgi\/image\/fit=contain,width=\d+,height=\d+/;
```

Comment:
In some light testing, I had difficulties with this regex pattern. The following does work for me:

```
const regex = /\/cdn-cgi\/image\/fit=contain,width=\d+,height=\d+/;
url = "/cdn-cgi/image/fit=contain,width=175,height=98/production/original_527737129.jpg"
console.log(url.replace(regex, (_match, p1, p2) => `/fit-in/${p1}/filters:quality(${p2})`));
```

I may be wrong so I wanted to ask.

Reply:
Let me see if I got your point correctly 
The regex in the function will fail for this type of url url = "/cdn-cgi/image/fit=contain,width=175,height=98/production/original_527737129.jpg"

The regex assumes that fit-in will be part of the url hence it does not matches 

The alternative does match but its output adds fit-in and quality parameters in the url where it may or may not originally be there . Output is also suspect .

I have added a playground with both the function 
https://tinyurl.com/ycksvczh

Reply:
One more point is that you might have got the url  from DD like here 
![image](https://github.com/stocktwits/serverless-image-handler/assets/117149188/1a3f243c-757f-47ee-b38d-fae56a583347)


However this does not include  the whole path this only includes the "key" for s3 bucket . 
If we look cloud watch for production

Reply:
Based on what you're saying, I believe there to be a misunderstanding. 

The `path` in the example you showed above is actually the wrong path for historical CDN-based image URLs.

Consider this URL, which was used when we leveraged Cloudflare as the solution to resize images:
```
https://charts.stocktwits-cdn.com/cdn-cgi/image/fit=contain,width=175,height=98/production/original_527737129.jpg
```
The above URL (or a derivative of it) does not contain the string `fit-in` as part of the path. 

This URL represents the same image using SIH:

```
https://sih-st-charts.stocktwits-cdn.com/fit-in/175x98/filters:quality(75)/production/original_527737129.jpg
```

It is possible that users are posting references to old URLs in their posts, so we'll need to correctly re-format them to use SIH.

@jaimegarcia has a good point in having those corrected in Core, but I would argue (and would lean more on) that we just handle it in SIH since there is a resourcing constraint.

Reply:
When FE sends the url to SIH it transforms it to this 

https://stocktwits.com/PeterCpenny/message/528175263


Request URL:

https://sih-st-charts.stocktwits-cdn.com/fit-in/500x0/filters:quality(75)/cdn-cgi/image/fit=contain,width=175,height=98/production/original_528174295.jpeg

here is an example in Chrome 


<img width="1725" alt="image" src="https://github.com/stocktwits/serverless-image-handler/assets/117149188/b5c4f51b-71ab-4040-b34c-1a0156c5ef59">

This code is trying to solve the above use case.

Reply:
The core sends the message like this 
"links": [
            {
                "title": "George (@Homer23) | Stocktwits",
                "url": "https://stocktwits.com/Homer23/message/528174295",
                "shortened_url": "https://stkt.co/V9UA3mVa",
                "shortened_expanded_url": "[stkt.co/V9UA3mVa](http://stkt.co/V9UA3mVa)",
                "description": "TRKA ....",
                "image": "https://charts.stocktwits-cdn.com/cdn-cgi/image/fit=contain,width=175,height=98/production/original_528174295.jpeg",
                "created_at": "2023-05-17T12:46:43Z",
                "video_url": null,
                "source": {
                    "name": "Stocktwits",
                    "website": "https://stocktwits.com/"
                }
            }
        ],

I believe FE removes the cdn TLD and adds https://sih-st-charts.stocktwits-cdn.com/fit-in/500x0/ in front